db/migrate/20240408132459_rename_delay_to_lag.rb

```diff
@@ -0,0 +1,6 @@
+class RenameDelayToLag < ActiveRecord::Migration[7.1]
+  def change
+    rename_column :relations, :delay, :lag
```

Comment:
❗ As discussed, this is a destructive migration and will need a downtime for deployment. As old code will still require data from the old DB state.

Reply:
What do you think about temporarily adding a virtual `delay` column, so that the old code can still work with the database model and we are able to migrate without a downtime:

```ruby
    rename_column :relations, :delay, :lag
    add_column :relations, :delay, :virtual, type: :integer, as: 'lag', stored: true
```

This would still fail when trying to write into the `delay` column (as it is generated), but would not die left and right. Not an optimal solution but a possibility

Reply:
For cloud, 14.0 will have a downtime associated to it. Since this doesn't improve the performance of the migration, I don't think it would be necessary (but open for thoughts @machisuji @ulferts ?)

Reply:
First of all, thank you very much for the thought and effort, @klaustopher!

@oliverguenther Are there any other migrations planned for 14.0?
Right now I only see two (20240311111957_enable_unaccent_extension.rb, 20240402065214_remove_ldap_tls_setting.rb), both of which would not require a downtime IMHO. 

So if no other migrations are being added, then I would be very much in favour of Klaus's solution here to avoid the necessity of a downtime.

If there are going to be new migrations which do require a downtime, we don't have to go the extra mile here, however.

Reply:
P.S. I didn't check the modules but so far only see `20240313102951_add_responsible_to_meeting.rb` there. But I guess that would need a downtime?

Reply:
Adding a new field would not need a downtime as adding new fields to the DB works while the old code is still live.

Reply:
I added it to the migration

Reply:
@machisuji yes, there are more migrations coming in (#14768). I guess that the ones in there don't require a downtime.